Mathlib/Probability/Distributions/Gaussian/Fernique.lean

```diff
@@ -0,0 +1,278 @@
+See `map_rotation_eq_self`. -/
+lemma map_rotation_eq_self_of_forall_strongDual_eq_zero
+    [SecondCountableTopology E] [CompleteSpace E]
+    (hμ : ∀ L : StrongDual ℝ E, μ[L] = 0) (θ : ℝ) :
+    (μ.prod μ).map (ContinuousLinearMap.rotation θ) = μ.prod μ := by
+  refine Measure.ext_of_charFunDual ?_
+  ext L
+  simp_rw [charFunDual_map, charFunDual_prod, charFunDual_eq_of_forall_strongDual_eq_zero hμ,
+    ← Complex.exp_add]
+  rw [← add_div, ← add_div, ← neg_add, ← neg_add]
+  congr 3
+  norm_cast
+  have h1 : (L.comp (.rotation θ)).comp (.inl ℝ E E)
+      = Real.cos θ • L.comp (.inl ℝ E E) - Real.sin θ • L.comp (.inr ℝ E E) := by
+    ext x
+    simp only [ContinuousLinearMap.coe_comp', Function.comp_apply, ContinuousLinearMap.inl_apply,
+      ContinuousLinearMap.rotation_apply, smul_zero, add_zero]
+    rw [← L.comp_inl_add_comp_inr]
+    simp [- neg_smul, sub_eq_add_neg]
+  have h2 : (L.comp (.rotation θ)).comp (.inr ℝ E E)
+      = Real.sin θ • L.comp (.inl ℝ E E) + Real.cos θ • L.comp (.inr ℝ E E) := by
+    ext x
+    simp only [ContinuousLinearMap.coe_comp', Function.comp_apply, ContinuousLinearMap.inr_apply,
+      ContinuousLinearMap.rotation_apply, smul_zero, zero_add, ContinuousLinearMap.add_apply,
+      ContinuousLinearMap.coe_smul', Pi.smul_apply, ContinuousLinearMap.inl_apply, smul_eq_mul]
+    rw [← L.comp_inl_add_comp_inr]
+    simp
+  rw [h1, h2]
+  simp only [ContinuousLinearMap.coe_sub', ContinuousLinearMap.coe_smul',
+    ContinuousLinearMap.coe_add']
+  rw [variance_sub, variance_smul, variance_add, variance_smul, variance_smul, covariance_smul_left,
+    covariance_smul_right, variance_smul, covariance_smul_left, covariance_smul_right]
+  · have h := Real.cos_sq_add_sin_sq θ
+    grind
+  all_goals exact (memLp_dual _ _ _ (by simp)).const_smul _
+
+end Rotation
+
+section Fernique
+
+variable [SecondCountableTopology E]
+
+/-- The convolution of a Gaussian measure `μ` and its map by `x ↦ -x` is centered. -/
+lemma integral_dual_conv_map_neg_eq_zero (L : StrongDual ℝ E) :
+    (μ ∗ (μ.map (ContinuousLinearEquiv.neg ℝ)))[L] = 0 := by
+  rw [integral_conv (by fun_prop)]
+  simp only [map_add]
+  calc ∫ x, ∫ y, L x + L y ∂μ.map (ContinuousLinearEquiv.neg ℝ) ∂μ
+  _ = ∫ x, L x + ∫ y, L y ∂μ.map (ContinuousLinearEquiv.neg ℝ) ∂μ := by
+    congr with x
+    rw [integral_add (by fun_prop) (by fun_prop)]
+    simp [- ContinuousLinearEquiv.coe_neg, integral_const, smul_eq_mul]
+  _ = ∫ x, L x ∂μ + ∫ y, L y ∂μ.map (ContinuousLinearEquiv.neg ℝ) := by
+    rw [integral_add (by fun_prop) (by fun_prop)]
+    simp
+  _ = 0 := by
+    rw [integral_map (by fun_prop) (by fun_prop)]
+    simp [integral_neg]
+
+/-- If `x ↦ exp (C * ‖x‖ ^ 2)` is integrable with respect to the centered Gaussian
+`μ ∗ (μ.map (ContinuousLinearEquiv.neg ℝ))`, then for all `C' < C`, `x ↦ exp (C' * ‖x‖ ^ 2)`
+is integrable with respect to `μ`. -/
+lemma integrable_exp_sq_of_conv_neg (μ : Measure E) [IsGaussian μ] {C C' : ℝ}
```

Comment:
is `conv` the recommended spelling for convolution? TIL!

Reply:
Yes, the definition is `Measure.conv`